### ISSUE TITLE
fix cached values in web containers

### ISSUE THREAD

**Issue by tylerflex** [flexcompute/tidy3d#1816]
TLDR: fix saving and loading `Job` to file when the simulation contains custom data.

`Job`, `Batch`, and `ComponentModeler` have `@properties` of `task_id`, `jobs`, and `batch`, respectively.

These get computed when the components are run, but we also wanted a way to save and load the computed properties from file. The fix I implemented originally was to introduce respective fields `task_id_cached`, `job_cached`, `batch_cached` , that are `None` originally, but then get saved when the `json()` of the component is created, eg when doing `to_file()`.  Then these fields can be loaded from file and the properties will use their values if they exist.

A problem occurs when the `Job.simulation` contains a custom dataset, eg custom medium. Then the json file writes the dataset path to file, but the data itself is lost / only stored in the corresponding actual `self.jobs[key].simulation` path.

So a solution is to more generally store and handle saving all cached properties to file? or we can try to solve this immediate problem.

To reproduce, add a custom medium to the smatrix notebook simulation and it will fail at `from_file()`

```py
fake_data = td.Structure(
    geometry=td.Box(
        center=(0,0,0),
        size=(1,1,1)
    ),
    medium=td.CustomMedium(
        permittivity=td.ScalarFieldDataArray(
            np.ones((1,1,1)),
            coords=dict(x=[0], y=[0], z=[0])
        )
    )
)


```
